### PR TITLE
chore: fixup tests that were failing for various reasons

### PR DESCRIPTION
**Pull request by mikecirioli**
[WIP] fixes test failures due to timing issues when promotion to staging had not completed before the test checked

### CODE REVIEW

**Comment by ccojocar** [comment]
@mikecirioli Please could check if this issue is addressed by this PR https://github.com/jenkins-x/bdd-jx/pull/9?

**Comment by mikecirioli** [comment]
@ccojocar not really, most of this is fixing additional issues in the BDD tests

**Comment by mikecirioli** [comment]
Removing the WIP label - a number of tests are failing due to underlying issues with the actual quickstarts, otherwise this PR addresses issues with checking for successful builds that were incorrectly failing tests.  I will validate and update the existing quickstarts and then do another round of test cleanup on this repo.

**Comment by mikecirioli** [comment]
@agentgonzo @ccojocar - please review

File: main.go [hunk 1]
```diff
@@ -125,14 +125,17 @@ func (t *Test) TheApplicationIsRunningInStaging(statusCode int) {
...
 		if u == "" {
 			return fmt.Errorf("No URL found for environment %s", key)
+			utils.LogInfo("still looking for app env info url")
 		}
 		return nil
 	}
-	err := RetryExponentialBackoff(time.Minute*10, f)
-	Expect(err).ShouldNot(HaveOccurred(), "get applications with a URL")
+	err := RetryExponentialBackoff(time.Minute * 10, f)
+	if err != nil {
```
Comment (agentgonzo) [inline main.go]: Why do you need to check for the error being non-nil before the `Expect`? Doesn't this defeat the point of the `Expect`?

File: main.go [hunk 2]
```diff
@@ -464,10 +484,9 @@ func CreateQuickstartTests(quickstartName string) bool {
 						T.CreatePullRequestAndGetPreviewEnvironment(200)
 					}
 
-					if T.DeleteApplications() {
+					if T.DeleteApps() {
 						By("deletes the application")
-						fullApplicationName := T.GetGitOrganisation() + "/" + T.ApplicationName
-						args = []string{"delete", "application", "-b", fullApplicationName}
+						args = []string{"delete", "application", "-b", T.AppName}
```
Comment (agentgonzo) [inline main.go]: I think with recent changes merged the other day, this needs to be `T.ApplicationName`

File: main.go [hunk 3]
```diff
@@ -464,10 +484,9 @@ func CreateQuickstartTests(quickstartName string) bool {
 						T.CreatePullRequestAndGetPreviewEnvironment(200)
 					}
 
-					if T.DeleteApplications() {
+					if T.DeleteApps() {
```
Comment (agentgonzo) [inline main.go]: I think this needs to be `T.DeleteApplications()`

File: main.go (hunk 1 above)
Comment (mikecirioli) [reply main.go]: i was seeing a NPE here occasionally, the best i could determine it was happening due to the `err.Error()` call in the message.  I can retest without those changes and see if i can reproduce.

Comment (agentgonzo) [reply main.go]: Yeah, that would make sense.
The code flow would be (I think):
1) Call `Expect(err)`, passing in `nil`. Store result as a temporary variable
2) evaluate `HaveOccurred()`
3) evaluate `err.Error()`    <---- NPE here
4) call `ShouldNot(...)` on the result of 1, passing in the result of 2, `"get applications with..."` and the result of 3

So AFAIUI, even if `ShouldNot` is a no-op, the `err.Error()` is getting evaluated and a NRE getting thrown.

**Comment by pmuir** [comment]
Merged based on @agentgonzo approval.

**Pull request merged**